src/context.cpp

```diff
@@ -596,61 +699,172 @@ namespace librealsense
             if ( profiles.size() > 0 )
             {
                 _dev->open( realdds_profiles );
+
             }
 
             software_sensor::open( profiles );
         }
 
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                std::shared_ptr< stream_profile_interface > prof, std::string name )
+        {
+            rs2_software_video_frame rs2_frame = {};
+
+            // prof parameter holds the real data, rs2_software_video_frame forces us to hold a pointer to it.
```

Comment:
Let me see if I can understand this:
* `rs2::frame` has an associated `rs2_stream_profile` which you can query (with `get_profile()`)
* however, we throw away the frame after we build the software frame
* therefore we throw away its associated profile
* so we need to copy it if we want to use it...?
But isn't the `prof` that we have a `shared_ptr< stream_profile_interface >`? If we already have it in hand, why not keep that around? Where is the usage that requires this, can you point to it?

In line 717, you basically take the pointer from `prof` and copy it to your own shared_ptr, but that does not add any reference count or anything -- if `prof` goes away, the pointer goes away. Just because you stored it somewhere doesn't mean it's valid...

Something's fishy here, I'd like to understand it.

Reply:
'prof' is a `stream_profile_interface` that originates from `get_active_streams` [here](https://github.com/IntelRealSense/librealsense/pull/11412/files/5d454bde748ad56ee2dff3fb22782b4e6e50b4c9#diff-776a5f8330d9fb807e058595e98557798a24587a21f080bb5d77542d50875acfR824). It is bound through the `on_data_available` lambda.

`rs2_software_video_frame` doesn't hold a `stream_profile_interface` directly, it holds a pointer to a `rs2_stream_profile` that, in turn, points to our `stream_profile_interface`. We need to allocate memory for the actual `rs2_stream_profile`, and we keep it alive while in the `syncer` queue.

Reply:
You're taking the pointer that's managed by a `shared_ptr` and copying it somewhere else. If the shared_ptr goes away, the pointer will no longer be valid. This cannot be good.

Reply:
As discussed, saving the `shared_ptr` in the `clone` field of the `rs2_stream_profile` to guaranty the pointer won't be freed. Added explanation in the comment.